### PR TITLE
Getting a customer by their id

### PR DESCRIPTION
**Pull request by danielKNS**
Getting a customer by their id

### CODE REVIEW

File: java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerController.java [hunk 1]
```diff
@@ -18,4 +20,9 @@ public void createNewCustomer(){
         customerService.createNewCustomer();
     }
 
+    @GetMapping("/customers/{id}")
+    public Customer gettingCustomerByid(@PathVariable Integer id){
+        System.out.println("getting customers by Id!!!");
+        return customerService.gettingCustomerById(id);
```
Comment (lgc13) [inline java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerController.java]: sysout deletion or logger please

File: java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerService.java [hunk 2]
```diff
@@ -29,4 +34,17 @@ public void createNewCustomer(){
...
+        String sql = "SELECT * FROM customer WHERE id = ?";
+        try{
+         //this is an example of Mapping: we are getting the string but it then reads essentially the string and places it the right fields(id etc.) for the object we have in java.
+            Customer customerId = jdbcTemplate.queryForObject(sql,new BeanPropertyRowMapper<>(Customer.class),id);
+            // queryForObject return the whole object(A specific object)
+            return customerId;
+        }catch (EmptyResultDataAccessException emptyResultDataAccessException){
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+    }
```
Comment (lgc13) [inline java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerService.java]: can we delete these comments, please?

Comment (lgc13) [reply java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerService.java]: can we add a log.error in the catch block?

additionally, can we add a message within the `ResponseStatusException`? (Second parameter)

**Review by lgc13 — APPROVED** [review]

File: java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerController.java (hunk 1 above)
Comment (danielKNS) [reply java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerController.java]: Done.

File: java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerService.java (hunk 2 above)
Comment (danielKNS) [reply java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerService.java]: The comments won't be deleted but i updated the ResponseStatusException and added the log.error

**Review by lgc13 — APPROVED** [review]

**Pull request merged**